10-gis.Rmd

```diff
@@ -23,65 +23,68 @@ library(gdalcubes)
 
 ## Introduction
 
-A defining feature of R is the way you interact with it:
-you type commands and hit `Enter` (or `Ctrl+Enter` if writing code in the source editor in RStudio\index{RStudio}) to execute them interactively.
-This way of interacting with the computer is called a command-line interface (CLI)\index{command-line interface} (see definition in the note below).
-CLIs are not unique to R.^[
-Other 'command-lines' include terminals for interacting with the operating system and other interpreted languages such as Python.
-Many GISs originated as a CLI:
-it was only after the widespread uptake of computer mice and high-resolution screens in the 1990s that GUIs\index{graphical user interface} became common.
-GRASS, one of the longest-standing GIS\index{GIS} programs, for example, relied primarily on command-line interaction before it gained a sophisticated GUI [@landa_new_2008].
-]
-In dedicated GIS\index{GIS} packages, by contrast, the emphasis tends to be on the graphical user interface (GUI)\index{graphical user interface}.
-You *can* interact with QGIS\index{QGIS}, SAGA\index{SAGA}, GRASS\index{GRASS} and gvSIG from system terminals and embedded CLIs\index{command-line interface}, but 'pointing and clicking' is the norm.
-This means many GIS\index{GIS} users miss out on the advantages of the command-line according to Gary Sherman, creator of QGIS\index{QGIS} [@sherman_desktop_2008]:
+A defining feature of [interpreted](https://en.wikipedia.org/wiki/Interpreter_(computing)) languages with an interactive console --- technically a read-eval-print loop (REPL) --- such as R is the way you interact with them:
+rather than relaying on a pointing and clicking on different parts of a screen, you type commands into the console and hit `Enter` each time you want a command to be executed.
+A common and effective workflow when using interactive development environments such as RStudio or VS Code is to type code into source files in a source editor and control interactive execution of the code with a shortcut such as `Ctrl+Enter`.
+In other words, R is typically controlled by a command-line interface (CLI)\index{command-line interface}.
+
+CLIs are not unique to R: most early computing environments relied on a command line 'shell' and it was only after the invention and widespread adoption of the computer mouse in the 1990s that graphical user interfaces (GUIs)\index{graphical user interface} became common.
+GRASS the longest-standing continuously-developed open source GIS\index{GIS} software, for example, relied on its command-line interface before it gained a GUI [@landa_new_2008].
```

Comment:
yeah, they celebrate their 40th anniversary this year in Prague which is amazing!!